src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.hxx

```diff
@@ -0,0 +1,276 @@
+public:
+
+  /**
+   * Constructor of the occ geometry class.
+   * \param [in] dimension  The dimension of this geometry.
+   * \param [in] fileprefix Prefix of a .brep file from which to extract an occ geometry.
+   * \param [in] name       The name to give this geometry.
+   */
+  t8_geometry_occ (int dimension, const char *fileprefix, const char *name);
+
+  /**
+   * Constructor of the occ geometry class.
+   * \param [in] dimension  The dimension of this geometry.
+   * \param [in] occ_shape  Occ shape geometry.
+   * \param [in] name       The name to give this geometry.
+   */
+  t8_geometry_occ (int dimension, const TopoDS_Shape occ_shape, 
+                   const char *name);
+
+  /** The destructor. 
+   * Clears the allocated memory.
+   */
+  virtual ~t8_geometry_occ ()
+  {
+    /* Nothing to do. */
+  }
+
+  /**
+   * Map a point in the reference space $$[0,1]^dimension$$ to $$\mathbb R^3$$
+   * \param [in]  cmesh      The cmesh in which the point lies.
+   * \param [in]  gtreeid    The global tree (of the cmesh) in which the reference point is.
+   * \param [in]  ref_coords  Array of \a dimension many entries, specifying a point in [0,1]^dimension.
+   * \param [out] out_coords  The mapped coordinates in physical space of \a ref_coords.
+   */
+  virtual void
+  t8_geom_evaluate (t8_cmesh_t cmesh,
+                    t8_gloidx_t gtreeid,
+                    const double *ref_coords,
+                    double out_coords[3]) const;
+
+  /**
+   * Compute the jacobian of the \a t8_geom_evaluate map at a point in the reference space $$[0,1]^dimension$$.
+   * \param [in]  cmesh      The cmesh in which the point lies.
+   * \param [in]  gtreeid    The global tree (of the cmesh) in which the reference point is.
+   * \param [in]  ref_coords  Array of \a dimension many entries, specifying a point in [0,1]^dimension.
+   * \param [out] jacobian    The jacobian at \a ref_coords. Array of size dimension x 3. Indices 3*i, 3*i+1, 3*i+2
+   *                          correspond to the i-th column of the jacobian (Entry 3*i + j is del f_j/del x_i).
+   */
+  virtual void
+  t8_geom_evalute_jacobian (t8_cmesh_t cmesh,
+                            t8_gloidx_t gtreeid,
+                            const double *ref_coords,
+                            double *jacobian) const;
+
+
+  /** Update a possible internal data buffer for per tree data.
+   * This function is called before the first coordinates in a new tree are
+   * evaluated. You can use it for example to load the vertex coordinates of the 
+   * tree into an internal buffer (as is done in the linear geometry).
+   * \param [in]  cmesh      The cmesh.
+   * \param [in]  gtreeid    The global tree.
+   */
+  virtual void 
+  t8_geom_load_tree_data (t8_cmesh_t cmesh,
+                          t8_gloidx_t gtreeid);
+  
+  /** Get an occ point from the occ_shape.
+   * \param [in] index      The index of the point in the occ_shape.
+   * \return                The occ point.
+   */
```

Comment:
Would it make sense to make all return values of the getters `const`?
The functions are marked const which indicates that the returned objects themselves should also not be modified.

Reply:
resolved in e092dc2a